Gems/ROS2/Code/Source/Lidar/PointCloudMessageBuilder.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <AzCore/std/optional.h>
+
+namespace ROS2
+{
+    class PointCloud2MessageBuilder
+    {
+    public:
+        PointCloud2MessageBuilder(const AZStd::string& frameId, builtin_interfaces::msg::Time timeStamp, size_t count);
+        PointCloud2MessageBuilder& AddField(const char* name, uint8_t dataType, size_t count = 1);
+        sensor_msgs::msg::PointCloud2 Build();
+
+    private:
+        size_t m_offset = 0U;
+        AZStd::optional<sensor_msgs::msg::PointCloud2> m_message;
```

Comment:
I made it a one time use object because the only scenario i had imagined in which the build method was called more than once would be a potential bug. I really do not have a strong opinion on this and will gladly change it.

Reply:
I appreciate the changes made! I would maybe consider some reset feature for message builder (+ persistent builder instead of creating builder per point cloud publish), but it is fine now and may be left like that.